### PR TITLE
Update to Hyper v0.14.*, Tokio v1.*

### PR DESCRIPTION
**Pull request by seanpianka**
Updating to latest versions of Hyper and Tokio

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,13 +22,12 @@ default = []
 all = []
 
 [dependencies]
-hyper = "0.13"
+hyper = { version = "0.14", default-features = false, features = ["full"] }
```
Comment (seanpianka) [inline Cargo.toml:25]: @rousan Do you know which features of hyper that Routerify is using? I specified full as a catch-all, but maybe we can pare it down.

Comment (rousan) [reply Cargo.toml:25]: We can keep it as it is for now.

**Comment by rousan** [comment]
Please send PR to develop branch.

**Comment by seanpianka** [comment]
> All the testcases are running?

```
$ cargo test
   Compiling routerify v2.0.0-beta-2 (/Users/sean/Development/routerify/routerify)
    Finished test [unoptimized + debuginfo] target(s) in 19.85s
     Running target/debug/deps/routerify-40bc3048c7a7be4f

running 5 tests
test helpers::tests::test_percent_decode_request_path ... ok
test regex_generator::tests::test_generate_common_regex_str_special_character ... ok
test regex_generator::tests::test_generate_common_regex_str_normal ... ok
test regex_generator::tests::test_generate_common_regex_str_star_globe ... ok
test regex_generator::tests::test_generate_common_regex_str_params ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/integration-d3ed8f185b2f16d9

running 2 tests
test can_perform_simple_get_request ... ok
test can_respond_with_data_from_scope_state ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests routerify

running 52 tests
test src/lib.rs - (line 48) ... ok
test src/lib.rs - (line 496) ... ok
test src/lib.rs - (line 576) ... ok
test src/lib.rs - (line 451) ... ok
test src/lib.rs - (line 199) ... ok
test src/lib.rs - (line 364) ... ok
test src/ext/request.rs - ext::request::RequestExt::params (line 12) ... ok
test src/lib.rs - (line 157) ... ok
test src/lib.rs - (line 182) ... ok
test src/lib.rs - (line 135) ... ok
test src/ext/request.rs - ext::request::RequestExt::remote_addr (line 65) ... ok
test src/ext/request.rs - ext::request::RequestExt::param (line 39) ... ok
test src/lib.rs - (line 332) ... ok
test src/lib.rs - (line 423) ... ok
test src/lib.rs - (line 255) ... ok
test src/lib.rs - (line 392) ... ok
test src/lib.rs - (line 218) ... ok
test src/lib.rs - (line 635) ... ok
test src/lib.rs - (line 681) ... ok
test src/router/builder.rs - router::builder::RouterBuilder (line 24) ... ok
test src/lib.rs - (line 284) ... ok
test src/lib.rs - (line 705) ... ok
test src/lib.rs - (line 736) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::post (line 61) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::post_with_info (line 88) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::post_with_info_with_path (line 177) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::post_with_path (line 149) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::pre (line 35) ... ok
test src/middleware/mod.rs - middleware::Middleware<B, E>::pre_with_path (line 122) ... ok
test src/middleware/post.rs - middleware::post::PostMiddleware<B, E>::new (line 59) ... ok
test src/middleware/post.rs - middleware::post::PostMiddleware<B, E>::new_with_info (line 88) ... ok
test src/middleware/pre.rs - middleware::pre::PreMiddleware<E>::new (line 44) ... ok
test src/route/mod.rs - route::Route (line 28) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::add (line 486) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::any (line 416) ... ok
test src/service/router_service.rs - service::router_service::RouterService (line 31) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::any_method (line 455) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::connect (line 325) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::get (line 115) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::delete (line 235) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::get_or_head (line 145) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::head (line 265) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::middleware (line 644) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::options (line 385) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::patch (line 355) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::post (line 175) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::put (line 205) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::trace (line 295) ... ok
test src/router/builder.rs - router::builder::RouterBuilder<B, E>::scope (line 527) ... ok
test src/router/mod.rs - router::Router (line 39) ... ok
test src/types/route_params.rs - types::route_params::RouteParams::get (line 32) ... ok
test src/types/route_params.rs - types::route_params::RouteParams::has (line 62) ... ok

test result: ok. 52 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by seanpianka** [comment]
I have also opened PRs for routerify-query and routerify-cors with similar updates.

**Comment by rousan** [comment]
@seanpianka, Looking for an active maintainer for this project as I'm stuck in some other projects. Would you be interested? If yes, you will be responsible to review PRs, fixes bugs etc.

**Comment by seanpianka** [comment]
> @seanpianka, Looking for an active maintainer for this project as I'm stuck in some other projects. Would you be interested? If yes, you will be responsible to review PRs, fixes bugs etc.

Hi @rousan, I'm willing to review raised issues and fix minor bugs, but if there's a safety issue in the codebase, you likely need to step in (as I'm not extremely familiar with those implementation details). 

Otherwise, I think a good plan going forward for this project would be to:
1. Close #15, #13
1. Merge #35, #38 
1. Implement #33 then close #6
1. Cut a new 2.0.0 release
1. Make the README more concise in feature descriptions, update benchmarks (as recent Warp more closely matches req/s)

From you, I will ask you to please provide:
1. A suggested approach for #7
2.  A 1-2 sentence description for #14
3. An answer to my question 😄: If #33 is implemented, Routerify will still have other code that currently depends on Hyper, correct?

**Comment by spikecodes** [comment]
This would be very useful, thanks!

**Comment by gsserge** [comment]
> Otherwise, I think a good plan going forward for this project would be to:
> 
> 1. Close #15, #13
> 2. Merge #35, #38
> 3. Implement #33 then close #6
> 4. Cut a new 2.0.0 release

@seanpianka do you think merging #35 and #38 would be enough for making 2.0.0 release? Let me know if you need some help with testing or anything. Would be nice to get the tokio/hyper update plus all the nice new features from master, like the new Error enum, into a stable release :)

**Comment by gsserge** [comment]
Btw, any particular reason for not merging this one yet?

**Comment by rousan** [comment]
Making a new beta release.

**Pull request merged**